### PR TITLE
Add miopentensile-new recipe for ROCm-5.0.0 . ROCm-5.1.0 release

### PR DESCRIPTION
**Pull request by srekolam**

### CODE REVIEW

File: var/spack/repos/builtin/packages/miopentensile/package.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
+# Copyright 2013-2022 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+
+from spack import *
+
+
+class Miopentensile(CMakePackage):
```
Comment (dtaller) [inline var/spack/repos/builtin/packages/miopentensile/package.py]: Hi. I was wondering why you do not have this inherit from ROCmPackage lib/spack/spack/build_systems/rocm.py, and then add related dependencies in that common place? 

So this line would become  `class Miopentensile(CMakePackage, ROCmPackage):` and some version information would go to that file?

I have not been involved in spack for a while. If there is something deficient/missing/incorrect in that approach for your application then that is ok, no problem.

**Comment by tldahlgren** [comment]
@alalazo Can you provide some insights into how to address the concretization issues raised by the package audits?

**Comment by srekolam** [comment]
This required the 5.1.0 updates for the dependents i believe which was not available during the time the PR was updated ..I can resync my branch again.

**Comment by tldahlgren** [comment]
> This required the 5.1.0 updates for the dependents i believe which was not available during the time the PR was updated ..I can resync my branch again.

Please do the rebase.

**Comment by srekolam** [comment]
@tldahlgren , i just did a rebase . Will wait for the results now.

**Comment by srekolam** [comment]
@tldahlgren  , i see the tests passing. Can you please let me know if this can get merged ?

**Review by tldahlgren — APPROVED** [review]
Confirmed sha256.

**Pull request merged**